pkg/irsa/capa/capa.go

```diff
@@ -299,9 +299,7 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	createOIDCProvider := func() error {
 		var identityProviderURLs []string
 		s3Endpoint := fmt.Sprintf("s3.%s.%s", s.Scope.Region(), key.AWSEndpoint(s.Scope.Region()))
-		if (key.IsV18Release(s.Scope.Release()) && !key.IsChina(s.Scope.Region())) || (s.Scope.MigrationNeeded() && !key.IsChina(s.Scope.Region())) {
-			identityProviderURLs = append(identityProviderURLs, util.EnsureHTTPS(cfDomain))
-		} else {
+		if !(key.IsV18Release(s.Scope.Release()) && !key.IsChina(s.Scope.Region())) || (s.Scope.MigrationNeeded() && !key.IsChina(s.Scope.Region())) {
```

Comment:
Just one thing, because I just saw it now. 

I don't think we need to check v18 release because that's a check for vintage only, we don't have those releases for CAPA.

The other thing is in China we must use the S3 Endpoint as identity provider because there's no CloudFront.